yabt/test_utils.py

```diff
@@ -31,11 +34,192 @@ def generate_random_dag(nodes, min_rank=0, max_rank=10, edge_prob=0.3):
     and edges can only go from nodes[i] -> nodes[j] if i < j
     (guaranteeing DAGness).
     """
-    g = networkx.DiGraph()
+    g = nx.DiGraph()
     g.add_nodes_from(nodes)
     for j in range(1, len(nodes)):
         rank = random.randint(min_rank, min(j, max_rank))
         g.add_edges_from((nodes[i], nodes[j])
                          for i in random.sample(range(j), k=rank)
                          if random.random() > edge_prob)
     return g
+
+
+def create_dag_eges(num, p):
+    """ Generate a directed acyclic graph
+    Parameters
+    ----------
+    num : int
+        The number of nodes.
+    p : float
+        Probability for edge creation.
+    ofname : str
+        Name of out dot file
+    """
+    edges = list()
+    for i in range(num):
+        for j in range(i+1, num):
+            edges.append((i, j))
+    e_num = num * (num - 1) // 2
+    e_num = min(e_num, int(e_num * p))
+    num = len(edges)
+    while num > e_num:
+        num = num - 1
+        edges.remove(edges[random.randint(0, num)])
+    return edges
+
+
+sort_test_cfg = [
+    {
+        "BuildTargets": [
+            {
+                "Name": "AptPackage", "Childs": ["AptPackage"], "MinChilds": 0
+            },
+            {
+                "Name": "CppGTest", "Childs": ["CppLib", "CppProg"],
+                "MinChilds": 1
+            },
+            {
+                "Name": "CppLib", "Childs": ["Proto", "AptPackage"],
+                "MinChilds": 1
+            },
+            {
+                "Name": "CppProg", "Childs": ["CppLib", "Proto", "AptPackage"],
+                "MinChilds": 1
+            },
+            {
+                "Name": "CustomInstaller", "Childs":
+                [
+                    "CppLib", "Proto", "CppProg", "CppGTest",
+                    "Python", "PythonPackage", "PythonTest"
+                ], "MinChilds": 3
+            },
+            {
+                "Name": "Proto", "Childs": ["Proto"], "MinChilds": 0
+            },
+            {
+                "Name": "Python", "Childs": ["Python", "PythonPackage"],
+                "MinChilds": 0
+            },
+            {
+                "Name": "PythonPackage", "Childs": ["PythonPackage"],
+                "MinChilds": 0
+            },
+            {
+                "Name": "PythonTest", "Childs": ["PythonPackage", "Python"],
+                "MinChilds": 2
+            }
+        ]
+    },
+    {
+        "Config": {
+            "MinNameLen": 2, "MaxNameLen": 5, "Steps": 100, "MaxStepNodes": 8
+        }
+    }
+]
+
+
+class CBuildTrgtTest:
```

Comment:
It is hard to understand what does this class do, try making the code clearer & add documentation